contracts/SSVNetwork.sol

```diff
@@ -672,12 +686,33 @@ contract SSVNetwork is UUPSUpgradeable, Ownable2StepUpgradeable, ISSVNetwork {
 
         operators[operatorId] = operator;
 
+        if (operatorsWhitelist[operatorId] != address(0)) {
+            delete operatorsWhitelist[operatorId];
+        }
+
         if (currentBalance > 0) {
             _transferOperatorBalanceUnsafe(operatorId, currentBalance.expand());
         }
         emit OperatorRemoved(operatorId);
     }
 
+    function _removeOperatorWhitelist(
+        uint64 operatorId,
+        Operator storage operator
+    ) private onlyOperatorOwner(operator) {
+        delete operatorsWhitelist[operatorId];
+        emit OperatorWhitelistRemoved(operatorId);
+    }
```

Comment:
Do we need a revert if its already empty?

Reply:
That would be useful mostly alerting the user. In terms of gas, if users use it without knowing if its set or not, then its more efficient to add the revert to not delete undefined value. but it users know the operator is whitelisted (via `Views.getOperatorById` `private` flag, then its more efficient to go an delete (current implementation).

Reply:
Resolving this and unresolved what we decided to go with